apps/web/app/lib/actions.ts

```diff
@@ -39,6 +39,46 @@ export async function authorizeUser(email: string, password: string) {
   return { success: true };
 }
 
+export async function createCourse(data: CourseFields) {
+  const { subject, code, title } = data;
+  await prisma.course.create({
+    data: {
+      subject,
+      code,
+      title,
+    },
+  });
+
+  return { success: true };
+}
```

Comment:
You should treat Server Actions as you would public-facing API endpoints, and ensure that the user is authorized to perform the action. 

So what you have would be:

```ts
export async function createCourse(data: CourseFields) {
  // dummy just to show usage
  const session = await getSession();
  if (!sessison) throw new Error("Unauthorized");

  const { subject, code, title } = data;
  await prisma.course.create({
    data: {
      subject,
      code,
      title,
    },
  });

  return { success: true };
}
```

Instead.

Read more about it [here.](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#authentication-and-authorization)